Source/AliveLibAO/Font.cpp

```diff
@@ -479,7 +480,7 @@ EXPORT int AliveFont::DrawString_41C360(int** ot, const char* text, __int16 x, _
 
         ++characterRenderCount;
 
-        offsetX += widthScaled + static_cast<short>(field_34_font_context->field_8_atlas_array[0].field_2_width * FP_GetExponent(scale));
+        offsetX += widthScaled + static_cast<short>(field_34_font_context->field_8_atlas_array[0].field_2_width * FP_GetDouble(scale));
```

Comment:
FP_GetDouble was supposed to be for debug printing of FixedPoints as float/double. Probably you want to convert width to a fixed point and then convert that fixed point to an integer, something like:

offsetX += widthScaled + FP_GetExponent(FP_FromInteger(field_34_font_context->field_8_atlas_array[0].field_2_width) * scale);

Assuming scale is an FP.

Reply:
Can do